File: egchat-mobile/android/app/src/main/java/com/egchat/app/modules/EGChatKeyboardViewManager.kt
package com.egchat.app.modules

import com.facebook.react.common.MapBuilder
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class EGChatKeyboardViewManager : SimpleViewManager<EGChatKeyboardView>() {
    override fun getName(): String = "EGChatKeyboardView"

    override fun createViewInstance(reactContext: ThemedReactContext): EGChatKeyboardView {
        return EGChatKeyboardView(reactContext)
    }

    @ReactProp(name = "text")
    fun setText(view: EGChatKeyboardView, text: String?) {
        view.setTextValue(text)
    }

    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any> {
        return mutableMapOf(
            "topChangeText" to mapOf("registrationName" to "onChangeText"),
            "topSubmit" to mapOf("registrationName" to "onSubmit")
        )
    }
}

File: egchat-mobile/ios/EGCHAT/EGChatKeyboardViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(EGChatKeyboardViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(text, NSString)
RCT_EXPORT_VIEW_PROPERTY(onChangeText, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onSubmit, RCTBubblingEventBlock)

@end

